docs/advanced-features/error-handling.md

```diff
@@ -0,0 +1,98 @@
+---
+description: Handle errors in your Next.js app.
+---
+
+# Error Handling
+
+This documentation explains how you can handle development, server-side, and client-side errors.
+
+## Handling Errors in Development
+
+When there is a runtime error during the development phase of your Next.js application, you will encounter an **overlay**. Fixing the error will automatically dismiss the overlay. It is only visible when the development server runs using `next dev`, `npm run dev`, or `yarn dev` and not in production.
```

Comment:
Do we need to explain what an "overlay" is? 

Basically: a modal that covers your current webpage

Or perhaps a screenshot?

Reply:
I think it will be a good addition to document it (especially for beginners). But I see you have explained it in a very concise way. I'll add that to it and a screenshot.

Reply:
Done ✅